src/DynamoCore/Logging/DynamoLogger.cs

```diff
@@ -194,6 +195,24 @@ public DynamoLogger(DebugSettings debugSettings, string logDirectory, Boolean is
             }
         }
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="DynamoLogger"/> class
+        /// with specified debug settings and directory where to write logs
+        /// </summary>
+        /// <param name="debugSettings">Debug settings</param>
+        /// <param name="logDirectory">Directory path where log file will be written</param>
+        /// <param name="isTestMode">Test mode is true or false.</param>
+        /// <param name="isCLIMode">CLI mode is true or false.</param>
+        public DynamoLogger(DebugSettings debugSettings, string logDirectory, Boolean isTestMode, Boolean isCLIMode)
+            :this(debugSettings, logDirectory, isTestMode)
+        {
+            cliMode = isCLIMode;
+            if (cliMode)
+            {
+                StartLoggingToConsoleAndFile(logDirectory);
```

Comment:
because you now are allowing arbitrary log directories, maybe inside `StartLoggingToConsoleAndFile` we should check that we have write access to the directory and that it exists?

Reply:
So nothing really changes, the log dir is the same as before.